components/kyma-environment-broker/internal/dto.go

```diff
@@ -224,26 +224,26 @@ type ERSContext struct {
 	Region                *string                            `json:"region,omitempty"`
 }
 
-func UpdateERSContext(provisioning, lastOperation ERSContext) ERSContext {
-	if lastOperation.SMOperatorCredentials != nil {
-		provisioning.SMOperatorCredentials = lastOperation.SMOperatorCredentials
+func UpdateERSContext(op, lastOperation ERSContext) ERSContext {
+	if op.SMOperatorCredentials == nil {
```

Comment:
the UpdateERSContext does not update field if was already set with other value before, is it intended?

Reply:
I realize now the naming here might be a bit confusing, it actually uses the current operation `op` by default and if any field is `nil`, it "inherits" the value from `lastOperation` which is populated as the first thing in each queue when the current one is still in `pending` status (so ignored by the `getLastOperation`).

Reply:
might be better to rename `op` to `currentOperation` and `lastOperation` to `previousOperation`